lib/spack/spack/environment/environment.py

```diff
@@ -1415,6 +1415,14 @@ def is_develop(self, spec):
         """Returns true when the spec is built from local sources"""
         return spec.name in self.dev_specs
 
+    def possible_dependencies(self):
+        """Get a set of names of possible dependencies for all roots."""
+        possible_deps = [s.name for s in self.user_specs]
+        for s in self.user_specs:
+            for d in spack.package_base.possible_dependencies(s).values():
```

Comment:
You haven't imported `spack.package_base`, and I was gonna suggest not to define public API for a rather niche use case and extend possible_dependencies(*specs) instead, but turns out that is the API already, making this loop redundant.

Just use `spack.package_base.possible_dependencies(*env.user_specs)`.

Reply:
Ah yes, I missed the leading `*` in the signature. This is much better, I was annoyed that I couldn't pass a list.